### PR TITLE
Getting a user's data by supplying a username

### PR DESCRIPTION
**Pull request by orhnsnmz**
The User.prototype.get function is broken for queries with particular usernames.
For example, if try to get the user "abcd" and there are multiple usernames which begin with "abcd", the function will return 'false'.

Also, the data returned from this query is not the same as querying for a user with a supplied id.
After supplying an id, you get more data, such as the media and follower counts, which does not happen if you search for a user with a given name/string.

The function, as it is now, will go through the results and look for an exact match between the supplied username and the users in the result set and then query the API again with the user's id, after finding a match.

### CODE REVIEW

File: src/instajam.js [hunk 1]
```diff
@@ -213,11 +213,19 @@
...
         // otherwise return nothing.
-        if (result.data && result.data.length === 1) {
-          result = result.data[0];
+        if (result.data) {    
+            var userIds = result.data;
+            result = false;
+
+            for (var i = 0; i < userIds.length; i++) {
+                if (userIds[i].username == id) {
+                    User.prototype.get(parseInt(userIds[i].id), callback);
```
Comment (mikefowler) [inline src/instajam.js]: Please use a radix for `parseInt`, a la `parseInt(userIds[i].id, 10)`. See [MDN for an explanation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt?redirectlocale=en-US&redirectslug=JavaScript%2FReference%2FGlobal_Objects%2FparseInt).

>  For this reason always specify a radix when using parseInt.

File: src/instajam.js [hunk 2]
```diff
@@ -213,11 +213,19 @@
...
+        // a perfect match, then query again with that user's ID
         // otherwise return nothing.
-        if (result.data && result.data.length === 1) {
-          result = result.data[0];
+        if (result.data) {    
+            var userIds = result.data;
+            result = false;
+
+            for (var i = 0; i < userIds.length; i++) {
+                if (userIds[i].username == id) {
```
Comment (mikefowler) [inline src/instajam.js]: Please change the comparison to a strict equals here.

**Comment by mikefowler** [comment]
@orhnsnmz, this looks like a good change. Please address the comments above and I'll happily merge this in.

**Comment by orhnsnmz** [comment]
@mikefowler sorry for the hold-up. I have committed your requested changes now. :)

**Comment by mikefowler** [comment]
Thanks, @orhnsnmz!

**Pull request merged**